CMakeLists.txt

```diff
@@ -193,37 +148,14 @@ set(INSTALL_TARGETS objc)
 # unconditionally (which means that it still defaults to C for .c files).
 set(ENV{CCC_OVERRIDE_OPTIONS} "x/TC x/Gm-")
 
-set_source_files_properties(
-	${libobjc_ASM_SRCS}
-	LANGUAGE C
-	COMPILE_FLAGS "${CMAKE_OBJC_FLAGS} -Xclang -x -Xclang assembler-with-cpp"
-)
-
-set_source_files_properties(
-	${libobjc_CXX_SRCS}
-	LANGUAGE CXX
-	COMPILE_FLAGS "${CMAKE_CXX_FLAGS}"
-)
-
-set_source_files_properties(
-	${libobjc_OBJC_SRCS}
-	COMPILE_FLAGS "${CMAKE_OBJC_FLAGS} -Xclang -x -Xclang objective-c"
-)
-
-set_source_files_properties(
-	${libobjc_OBJCXX_SRCS}
-	COMPILE_FLAGS "${CMAKE_OBJC_FLAGS} -Xclang -x -Xclang objective-c++"
-)
-
-
 #
 # C++ Runtime interaction
 #
 
 
 function(test_cxx CXX_RUNTIME_NAME IS_STDLIB)
 	set(CXX_RUNTIME_NAME_FULL "${CMAKE_SHARED_LIBRARY_PREFIX}${CXX_RUNTIME_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX}")
-	find_library(CXX_RUNTIME_LIB NAMES ${CXX_RUNTIME_NAME_FULL} ${CXX_RUNTIME_NAME})
+	find_library(CXX_RUNTIME_LIB NAMES ${CXX_RUNTIME_NAME_FULL})
```

Comment:
Why was this parameter removed? When building for Android, I previously got the following log output:
```
-- Testing C++ interop
-- Testing <ndk-path>/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/lib/arm-linux-androideabi/libc++abi.a as the C++ runtime library
-- Using <ndk-path>/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/lib/arm-linux-androideabi/libc++abi.a as the C++ runtime library
```

Building the branch will not find libc++abi.a, and instead go the "Testing C++ standard library" route.

Reply:
I wanted to change to just depending on the C++ standard library.  We don't gain much by avoiding that as a dependency and we lose the ability to depend on C++ standard library functions.

Reply:
I think I didn't make all of those changes.

We originally linked to the C++ runtime directly for two reasons:

 - We needed to know what it was to accommodate subtle ABI differences.
 - We wanted to build without any C++ support

The first of these doesn't matter anymore - we dynamically detect the runtime.

The second is no longer something that I care about.  We use C++ features in non-optional code paths and I'd rather be able to depend on a C++ standard library existing.  As such, I'd like to rip out all of the runtime detection logic and just link as C++ to pull in a standard library implementation.

Reply:
Ah I see, thanks for the explanation. I was just concerned that the logging change indicated an issue, but it sounds like it’s expected. I did just run our Android app with a new library build from this branch and it’s working as expected.